src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagService.cs

```diff
@@ -48,18 +56,42 @@ public async Task<AddCustomTagResponse> AddCustomTagAsync(IEnumerable<CustomTagE
             _customTagEntryValidator.ValidateCustomTags(customTags);
 
             Dictionary<long, CustomTagStoreEntry> addedTags = new Dictionary<long, CustomTagStoreEntry>();
-            foreach (var tag in customTags)
+            foreach (var customTag in customTags)
             {
                 try
                 {
                     CustomTagStatus initStatus = CustomTagStatus.Reindexing;
-                    long key = await _customTagStore.AddCustomTagAsync(tag.Path, tag.VR, tag.Level, initStatus);
-                    CustomTagStoreEntry storeEntry = new CustomTagStoreEntry(key, tag.Path, tag.VR, tag.Level, initStatus);
+
+                    // we only support top level dicomtag right now, will support embeded dicomtag soon.
+                    DicomTag[] tags;
+                    if (!_dicomTagParser.TryParse(customTag.Path, out tags, supportMultiple: false))
+                    {
+                        // not a valid dicom tag path
+                        throw new CustomTagEntryValidationException(
+                            string.Format(CultureInfo.InvariantCulture, DicomCoreResource.InvalidCustomTag, customTag));
+                    }
+
+                    DicomTag tag = tags[0];
+                    string path = tag.GetPath();
+
+                    string vr = customTag.VR;
+
+                    // when VR is not specified for standard tag,
+                    if (!tag.IsPrivate && tag.DictionaryEntry != DicomDictionary.UnknownTag)
+                    {
+                        if (string.IsNullOrWhiteSpace(vr))
+                        {
+                            vr = tag.GetDefaultVR()?.Code;
+                        }
+                    }
+
+                    long key = await _customTagStore.AddCustomTagAsync(path, vr, customTag.Level, initStatus);
```

Comment:
Would it make more sense for the `AddCustomTagAsync` to return a `CustomTagStoreEntry` with the `key` populated?